### PR TITLE
Add flag to ignore replay data message type

### PR DESCRIPTION
**Pull request by HMiladi**

### CODE REVIEW

File: src/app.cpp [hunk 1]
```diff
@@ -58,7 +58,8 @@ void transform(
 	bool tecmp_only,
 	light_packet_interface packet_interface,
 	light_packet_header packet_header,
-	const uint8_t* packet_data
+	const uint8_t* packet_data,
+	bool ignore_replay_data
```
Comment (kayoub5) [inline src/app.cpp:62]: Put this argument next to the `tecmp_only` argument

File: src/app.cpp [hunk 2]
```diff
@@ -74,6 +75,10 @@ void transform(
 	else {
 		// tecmp packet
 		while (res == 0) {
+			if (ignore_replay_data && header.message_type == TECMP_TYPE_REPLAY_DATA) {
```
Comment (kayoub5) [inline src/app.cpp:78]: You don't have to do this check inside the while loop, replay data don't get mixed with other message types

File: src/app.cpp [hunk 3]
```diff
@@ -167,6 +172,7 @@ int main(int argc, char* argv[]) {
 	args::Positional<std::string> inarg(parser, "infile", "Input File", args::Options::Required);
 	args::Positional<std::string> outarg(parser, "outfile", "Output File", args::Options::Required);
 	args::Flag tecmp_only(parser, "tecmp-only", "Only process TECMP packets, drop others", { "tecmp-only" }, args::Options::Single);
+	args::Flag ignore_replay_data(parser, "ignore-replay-data", "Ignore message type replay data", { "ignore-replay-data" }, args::Options::Single);
```
Comment (kayoub5) [inline src/app.cpp:175]: "ignore" means do nothing with them (pass them as is), here you are "dropping" them.
```suggestion
	args::Flag drop_replay_data(parser, "drop-replay-data", "Drop replay data messages", { "drop-replay-data" }, args::Options::Single);
```

**Comment by kayoub5** [comment]
CI failed

File: src/app.cpp (hunk 2 above)
Comment (HMiladi) [reply src/app.cpp:78]: You mean that the check should be done inside `tecmp_next` method?

Comment (kayoub5) [reply src/app.cpp]: it should be done just before the `while (res == 0)` check

File: src/app.cpp [hunk 4]
```diff
@@ -64,14 +65,15 @@ void transform(
 	tecmp_header header;
 	uint8_t* data;
 	int res = tecmp_next(packet_data, packet_header.captured_length, &iterator, &header, &data);
+	drop_replay_data = drop_replay_data && header.message_type == TECMP_TYPE_REPLAY_DATA;
```
Comment (kayoub5) [inline src/app.cpp:68]: wrong logic here, create a new variable, otherwise not all replay will be dropped

**Comment by kayoub5** [comment]
missing unit test, please make sure uploaded pcap have no confidential information

File: src/app.cpp (hunk 4 above)
Comment (HMiladi) [reply src/app.cpp]: done

**Comment by HMiladi** [comment]
@kayoub5 LGTY?

**Review by sensokame — APPROVED** [review]

**Pull request merged**